packaging_automation/tests/test_prepare_release.py

```diff
@@ -0,0 +1,106 @@
+        assert file_includes_line(TEST_BASE_PATH, MULTI_EXTENSION_OUT, " 10.2devel")
+        assert count_line_in_file(TEST_BASE_PATH, MULTI_EXTENSION_SQL,
+                                  "ALTER EXTENSION citus UPDATE TO '10.2-1';") == 2
+        assert file_includes_line(TEST_BASE_PATH, CONFIG_PY, "MASTER_VERSION = '10.2'")
+        assert file_includes_line(TEST_BASE_PATH, CONFIGURE_IN, "AC_INIT([Citus], [10.2devel])")
+        assert file_includes_line(TEST_BASE_PATH, CONFIGURE, "PACKAGE_VERSION='10.2devel'")
+        assert file_includes_line(TEST_BASE_PATH, CONFIGURE, "PACKAGE_STRING='Citus 10.2devel'")
+        assert file_includes_line(TEST_BASE_PATH, CONFIGURE,
+                                  r"\`configure' configures Citus 10.2devel to adapt to many kinds of systems.")
+        assert file_includes_line(TEST_BASE_PATH, CONFIGURE,
+                                  '     short | recursive ) echo "Configuration of Citus 10.2devel:";;')
+        assert file_includes_line(TEST_BASE_PATH, CONFIGURE, "PACKAGE_VERSION='10.2devel'")
+        assert os.path.exists(f"{TEST_BASE_PATH}/{update_release_return_value.upgrade_path_sql_file}")
+        run(f"git checkout {MAIN_BRANCH}")
+    finally:
+        clear_env()
+
+
+def test_patch_release():
+    initialize_env()
+    os.chdir("citus")
+    try:
+        update_release(github_token=github_token, project_name="citus", project_version="10.2.0",
+                       main_branch=MAIN_BRANCH,
+                       earliest_pr_date=datetime.strptime('2021.03.25 00:00', '%Y.%m.%d %H:%M'),
+                       exec_path=TEST_BASE_PATH, is_test=True)
+
+        update_release_return_value = update_release(
+            github_token=github_token, project_name="citus", project_version="10.2.1",
+            main_branch=MAIN_BRANCH,
+            earliest_pr_date=datetime.strptime('2021.03.25 00:00', '%Y.%m.%d %H:%M'),
+            exec_path=TEST_BASE_PATH, is_test=True)
+        assert file_includes_line(TEST_BASE_PATH, MULTI_EXTENSION_OUT, " 10.2.1")
+        assert file_includes_line(TEST_BASE_PATH, CONFIGURE_IN, "AC_INIT([Citus], [10.2.1])")
+        assert file_includes_line(TEST_BASE_PATH, CONFIGURE, "PACKAGE_VERSION='10.2.1'")
+        assert file_includes_line(TEST_BASE_PATH, CONFIGURE, "PACKAGE_STRING='Citus 10.2.1'")
+        assert file_includes_line(TEST_BASE_PATH, CONFIGURE,
+                                  r"\`configure' configures Citus 10.2.1 to adapt to many kinds of systems.")
+        assert file_includes_line(TEST_BASE_PATH, CONFIGURE,
+                                  '     short | recursive ) echo "Configuration of Citus 10.2.1:";;')
+        assert file_includes_line(TEST_BASE_PATH, CONFIGURE, "PACKAGE_VERSION='10.2.1'")
+        run(f"git checkout {MAIN_BRANCH}")
+
+    finally:
+        clear_env()
+
+
+def clear_env():
+    if os.path.exists("../citus"):
+        os.chdir("..")
+        run("chmod -R 777 citus")
+        run("sudo rm -rf citus")
```

Comment:
Why do we need to set file modes before deleting them?

```suggestion
        run("sudo rm -rf ../citus")
```

Reply:
Fixed

Reply:
I do not agree that this one is actually fixed yet. We still have one extra line.

Reply:
See https://github.com/citusdata/tools/pull/91/files#r641363744 for my suggestion in the renamed file.

Reply:
Fixed